### DISCUSSION TITLE
Add Feature to Deploy & Manage Intrusion Detection/Prevention System like CrowdSec

### DISCUSSION THREAD

**Discussion by joglomedia** [vitodeploy/vito#307] [Ideas]
Hi there,

I'm just starting to love Vito!

Anyway, cyberattacks, such as brute force attacks, spam bots, and other malicious activities, are becoming increasingly sophisticated and frequent. Installing an intrusion detection and prevention (IDS/IPS) system can provide a significant layer of protection for our server. This proactive measure can help prevent potential downtime, data breaches, and financial losses.

Is there a plan to integrate an IDS/IPS system into VitoDeploy? We might want to consider CrowdSec as a possible option.

Regards,

**Comment by rdp77** [comment]
Yes, I also agree that CrowdSec should be added as an optional security feature alongside Fail2Ban. Users can choose between Fail2Ban and CrowdSec according to their security needs.

CrowdSec will provide modern threat detection and collaborative IP reputation assessment, whilst Fail2Ban will remain available for users who prefer the existing lightweight approach.
